src/lib/userStorage/UserStorageClass.js

```diff
@@ -937,85 +861,7 @@ export class UserStorage {
    * @throws Error if profile is invalid
    */
   async setProfile(profile: UserModel, update: boolean = false): Promise<> {
-    if (profile && !profile.validate) {
-      profile = getUserModel(profile)
-    }
-
-    const { errors, isValid } = profile.validate(update)
-
-    if (!isValid) {
-      logger.warn(
-        'setProfile failed',
-        'Fields validation failed',
-        new Error('setProfile failed: Fields validation failed'),
-        { errors, category: ExceptionCategory.Human },
-      )
-
-      throw errors
-    }
-
-    if (profile.avatar) {
-      profile.smallAvatar = await resizeImage(profile.avatar, 50)
-    }
-
-    /**
-     * Checking fields to save which changed, even if have undefined value (for example empty mobile input field return undefined).
-     */
-    const fieldsToSave = keys(this.profileSettings).filter(key => key in profile)
-
-    /**
-     * Forming a new object of profile fields those have changed with default value if fields have undefined.
-     */
-    const profileWithDefaults = defaults(
-      Object.assign({}, ...fieldsToSave.map(field => ({ [field]: profile[field] }))),
-
-      /**
-       * Picked only those fields that have changed for setting default value if new field value equal undefined.
-       */
-      pick(this.profileDefaults, fieldsToSave),
-    )
-    const results = await Promise.all(
-      fieldsToSave.map(async field => {
-        let isPrivate
-
-        try {
-          isPrivate = get(this.profileSettings, `[${field}].defaultPrivacy`, 'private')
-
-          if (update) {
-            isPrivate = await this.getFieldPrivacy(field)
-          }
-
-          return await this.setProfileField(field, profileWithDefaults[field], isPrivate)
-        } catch (e) {
-          logger.warn('setProfile field failed:', e.message, e, {
-            field,
-            value: profileWithDefaults[field],
-            isPrivate,
-          })
-
-          return { err: `failed saving field ${field}` }
-        }
-      }),
-    )
-
-    const gunErrors = results.filter(ack => ack && ack.err).map(ack => ack.err)
-
-    if (gunErrors.length <= 0) {
-      return true
-    }
-
-    logger.error(
-      'setProfile partially failed',
-      'some of the fields failed during saving',
-      new Error('setProfile: some fields failed during saving'),
-      {
-        errCount: gunErrors.length,
-        errors: gunErrors,
-        strErrors: JSON.stringify(gunErrors),
-      },
-    )
-
-    throw gunErrors
+    return this.userProfileStorage.setProfile(profile, update)
```

Comment:
why did you remove all this code?
did you understand what it does before deciding to remove it?